### PR TITLE
Floor Value in Binary Search Tree(BST)

### PR DESCRIPTION
**Pull request by Mrjoy832**
## Related Issue
- #460 

## Proposed Changes
- 1. Added Code and Output to find Floor value in BST
- 2. Updated readme file of  LearnCPP/T-Tree 

## Additional Info
- Added the code with minimal approach of O(h) time complexity.

## Checklist

- [X] ✅ My code follows the code style of this project.
- [X] 📝 My change requires a change to the README.
- [X] 🎀 I have updated the README accordingly.
- [X] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [X] ✨ I have added tests to cover my changes.
- [X] 🚩 All new and existing tests passed.
- [X] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
---

## Output Screenshots
| Output #1    
| key=14 and 7  | 
Floor1 | |Output #2  |
| For key =1    |
floor2

### CODE REVIEW

**Comment by Mrjoy832** [comment]
Will be updating readme and describition soon

**Comment by Mrjoy832** [comment]
@siddhi-244 kindly look into this PR

File: T-Tree/B-BinarySearchTree/FloorValueBST.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        else{ // if node value is greater than key Go to Left
+            root=root->left;
+        }
+    }
+    return floor;
+}
+
+int main(){
+    struct node*p=new node(10);
+    p->left=new node(5);
```
Comment (siddhi-244) [inline T-Tree/B-BinarySearchTree/FloorValueBST.cpp:40]: Try to take the tree as input !

Comment (Mrjoy832) [reply T-Tree/B-BinarySearchTree/FloorValueBST.cpp:40]: @siddhi-244 , Updated readme taking this tree as input..Kindly have a look

**Comment by Mrjoy832** [comment]
Hey! @Lakhankumawat ,

**1st Merged PR** -> https://github.com/Lakhankumawat/LearnCPP/pull/726
**2nd Merged PR** -> https://github.com/Lakhankumawat/LearnCPP/pull/517

File: T-Tree/B-BinarySearchTree/FloorValueBST.cpp (hunk 1 above)
Comment (siddhi-244) [reply T-Tree/B-BinarySearchTree/FloorValueBST.cpp]: Take tree as input in cpp file

Comment (Mrjoy832) [reply T-Tree/B-BinarySearchTree/FloorValueBST.cpp]: Already in cpp file  the same tree is taken as input ...

Comment (siddhi-244) [reply T-Tree/B-BinarySearchTree/FloorValueBST.cpp]: The tree is not taken as input here ! You have to take it as a input from the user !

**Comment by Mrjoy832** [comment]
@siddhi-244 , added the tree as a input from user please check

**Review by siddhi-244 — APPROVED** [review]
LGTM

**Review by Lakhankumawat — APPROVED** [review]
LGTM 🌟🚩

**Comment by Lakhankumawat** [comment]
CGC  ( Continuous good contribution  ) , Your PR is promoted to level two 🥂

**Comment by Lakhankumawat** [comment]
@all-contributors please add @Mrjoy832  code ,

**Pull request merged**